OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListActivity.java

```diff
@@ -48,13 +59,92 @@ public static Intent newIntent() {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
         initWhatsNew();
-        configureDrawer();
-        setTitle(getString(R.string.label_my_courses));
+        if (!environment.getConfig().isTabsLayoutEnabled()) {
+            addDrawer();
+        } else {
+            addClickListenerOnProfileButton();
+        }
+        setTitle(R.string.label_my_courses);
         environment.getAnalyticsRegistry().trackScreenView(Analytics.Screens.MY_COURSES);
     }
 
+    private void addClickListenerOnProfileButton() {
+        findViewById(R.id.toolbar_profile_image).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                environment.getRouter().showUserProfile(MyCoursesListActivity.this, loginPrefs.getUsername());
+            }
+        });
+    }
+
+    @Override
+    protected void configureActionBar() {
+        ActionBar bar = getSupportActionBar();
+        if (bar != null) {
+            final boolean showHomeBtn = !environment.getConfig().isTabsLayoutEnabled();
+            bar.setDisplayShowHomeEnabled(showHomeBtn);
+            bar.setDisplayHomeAsUpEnabled(showHomeBtn);
+            bar.setIcon(android.R.color.transparent);
+        }
+    }
+
+    @Override
+    protected int getToolbarLayoutId() {
+        return environment.getConfig().isTabsLayoutEnabled() ?
+                R.layout.toolbar_with_profile_button :
+                super.getToolbarLayoutId();
+    }
+
+    @Override
+    public void setTitle(int titleId) {
+        setTitle(getResources().getString(titleId));
+    }
+
+    @Override
+    public void setTitle(CharSequence title) {
+        final View toolbar = findViewById(R.id.toolbar);
+        if (toolbar != null) {
+            final View titleView = toolbar.findViewById(R.id.toolbar_title_view);
+            if (titleView != null && titleView instanceof TextView) {
+                ((TextView) titleView).setText(title);
+                super.setTitle(null);
```

Comment:
This is what is causing the accessibility problem i.e. screen name not getting announced upon entering the screen.